### PR TITLE
fix(ui) - glossary feedbacks

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Fixes following points

- Allow to deselect owner in Glossary and Glossary Terms. (For glossary deselecting owner will clear the owner. For glossary term, on clearing the owner if there is an owner of Glossary, it will retain the owner of Glossary as owner of Glossary Term)
- Show Display name instead of Name on Glossary Left Panel - https://github.com/open-metadata/OpenMetadata/issues/11407#issuecomment-1534028499

I worked on ... because ...

Screenshot 2023-05-04 at 11 33 47 AM
Screenshot 2023-05-04 at 11 56 23 AM

#
### Type of change:

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    { name: 'Reference 1', endpoint: 'http://example.com/1' },
+    { name: 'Reference 2', endpoint: 'http://example.com/2' },
+  ];
+
+  const defaultProps = {
+    references,
+    isVisible: true,
+    onClose: mockOnClose,
+    onSave: mockOnSave,
+  };
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx:31]: Keep it outside the describe blocks to have clean tests blocks

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx:35]: We don't need this as we have set this globally in jest config

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  it('renders correctly', () => {
+    render(<GlossaryTermReferencesModal {...defaultProps} />);
+
+    expect(screen.getByText('label.reference-plural')).toBeInTheDocument();
+    expect(screen.getByText('label.add')).toBeInTheDocument();
+    expect(screen.getByText('label.cancel')).toBeInTheDocument();
+    expect(screen.getByText('label.save')).toBeInTheDocument();
+  });
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx:44]: Can we add a few more tests like interactions / Saving / Selecting actions in the modal?
Make sure we are verifying API call or prop method call here

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+describe('GlossaryOverviewTab', () => {
+  const onUpdate = jest.fn();
+  const selectedData = MOCKED_GLOSSARY_TERMS[0];
+  const permissions = MOCK_PERMISSIONS;
+  const isGlossary = true;
+
+  beforeEach(() => {
+    onUpdate.mockClear();
+  });
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx:41]: remove

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  it('renders the component', async () => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx:43]: - Consider passing empty value to reviewer, Synonyms, Tags, Terms and verify no data placeholders
- Add button if data isn't there
- Edit button if data is there

File: openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryLeftPanel/GlossaryLeftPanel.component.tsx [hunk 6]
```diff
@@ -54,7 +54,7 @@ const GlossaryLeftPanel = ({ glossaries }: GlossaryLeftPanelProps) => {
         ...acc,
         {
           key: glossary.name,
-          label: glossary.name,
+          label: glossary.displayName || glossary.name,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryLeftPanel/GlossaryLeftPanel.component.tsx:57]: we can use getEntityName method here

Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryLeftPanel/GlossaryLeftPanel.component.tsx]: Fixed

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx (hunk 5 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryOverviewTab.test.tsx:43]: These cases mentioned here are written in individual components. These are unit tests.

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx (hunk 3 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermReferencesModal.test.tsx:40]: Test Added

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**